src/troute-network/troute/AbstractNetwork.py

```diff
@@ -474,20 +474,20 @@ def initialize_routing_scheme(self,):
         routing_type = [run_hybrid, use_topobathy, run_refactored]
 
         _routing_scheme_map = {
-            MCOnly: [False, False, False],
-            MCwithDiffusive: [True, False, False],
-            MCwithDiffusiveNatlXSectionNonRefactored: [True, True, False],
-            MCwithDiffusiveNatlXSectionRefactored: [True, True, True],
+            MCOnly: [[False, False, False]],
+            MCwithDiffusive: [[True, False, False],[True, True, False]],
+            MCwithDiffusiveNatlXSectionNonRefactored: [[True, True, False]],
```

Comment:
I'm confused here, it looks like you added a `[True, True, False]` option (line 478), but that already exists as `MCwithDiffusiveNatlXSectionNonRefactored` (line 479)

Reply:
It is true but MCwithDiffusive was mainly written for using topo data while 'use_topobathy'=False.   So, when 'use_topobathy'=False, MCwithDiffusive throws errors in many cases (e.g., L221).   So I guess then the fix can be that we keep _routing_scheme_map as before, but keep the changes I made in AbstractNetwork.py.     Or, we can move L216~L237 to somewhere under class MCwithDiffusiveNatlXSectionNonRefactored?

Reply:
Also, RouteLink file doesn't have column name 'mainstem', we cannot use def _fill_in_missing_topo_data of AbstractRouting.py for NHD.   Furthermore, topobathy data isn't available anymore for NHD.  So, unless communities try to make their own topodata for NHD,  'MConly' and 'MCwithDiffusive (with use_topobathy=False) are only options to choose for NHD.